sphinxcontrib/test_reports/functions/__init__.py

```diff
@@ -1,16 +1,28 @@
+import re
+
+
 def tr_link(app, need, needs, test_option, target_option, *args, **kwargs):
     if test_option not in need:
         return ""
-    test_opt = need[test_option]
+    test_option_value = need[test_option]
 
     links = []
     for need_target in needs.values():
+        # Skip linking to itself
+        if need_target["id"] == need["id"]:
+            continue
+        
         if target_option not in need_target:
             continue
 
-        if (
-            test_opt == need_target[target_option] and test_opt is not None and len(test_opt) > 0  # fmt: skip
-        ):
-            links.append(need_target["id"])
+        if test_option_value is not None and len(test_option_value) > 0:
+            target_option_value = need_target[target_option]
+            if target_option_value is not None and len(target_option_value) > 0:
+                if test_option_value == target_option_value:
+                    links.append(need_target["id"])
+                # if the test option value has a *, use regex matching
+                elif "*" in test_option_value:
+                    if re.match(test_option_value, target_option_value):
```

Comment:
AFAIK `re.match` compiles the pattern with each call (pattern here is `test_option_value`)
As `test_option_value` is not changed during the for loop, we should maybe compile it in advance, near to the line `test_option_value = need[test_option]`:
`test_pattern = re.compile(test_option_value)`

Then we also could call `test_pattern.match(target_option_value)` always, not only if a `*` is in it.
Will save some performance :)

Reply:
Nice finding 😀